t/t2409-sdexec.t

```diff
@@ -0,0 +1,282 @@
+	    systemctl --user list-units --type=service >name.out &&
+	grep t2409-funfunfun name.out
+'
+test_expect_success 'sdexec can set unit Description property' '
+	cat >desc.exp <<-EOT &&
+	Description=fubar
+	EOT
+	$sdexec -r 0 \
+	    --setopt=SDEXEC_NAME=t2409-desc.service \
+	    --setopt=SDEXEC_PROP_Description=fubar \
+	    systemctl --user show --property Description \
+	        t2409-desc.service >desc.out &&
+	test_cmp desc.exp desc.out
+'
+test_expect_success 'sdexec can set unit MemoryHigh property' '
+	cat >memhi.exp <<-EOT &&
+	MemoryHigh=1073741824
+	EOT
+	$sdexec -r 0 \
+	    --setopt=SDEXEC_NAME="t2409-memhi.service" \
+	    --setopt=SDEXEC_PROP_MemoryHigh=1G \
+	    systemctl --user show --property MemoryHigh \
+	        t2409-memhi.service >memhi.out &&
+	test_cmp memhi.exp memhi.out
+'
+test_expect_success 'sdexec can set unit MemoryMax property' '
+	cat >memmax.exp <<-EOT &&
+	MemoryMax=infinity
+	EOT
+	$sdexec -r 0 \
+	    --setopt=SDEXEC_NAME="t2409-memmax.service" \
+	    --setopt=SDEXEC_PROP_MemoryMax=infinity \
+	    systemctl --user show --property MemoryMax \
+	        t2409-memmax.service >memmax.out &&
+	test_cmp memmax.exp memmax.out
+'
+test_expect_success 'sdexec can set unit MemoryMin property' '
+	cat >memmin.exp <<-EOT &&
+	MemoryMin=1048576
+	EOT
+	$sdexec -r 0 \
+	    --setopt=SDEXEC_NAME="t2409-memmin.service" \
+	    --setopt=SDEXEC_PROP_MemoryMin=1M \
+	    systemctl --user show --property MemoryMin \
+	        t2409-memmin.service >memmin.out &&
+	test_cmp memmin.exp memmin.out
+'
+test_expect_success 'sdexec can set unit AllowedCPUs property' '
+	cat >allowedcpus.exp <<-EOT &&
+	AllowedCPUs=0 57-99 127
+	EOT
+	$sdexec -r 0 \
+	    --setopt=SDEXEC_NAME="t2409-allowedcpus.service" \
+	    --setopt=SDEXEC_PROP_AllowedCPUs="0,57-99,127" \
+	    systemctl --user show --property AllowedCPUs \
+	        t2409-allowedcpus.service >allowedcpus.out &&
+	test_cmp allowedcpus.exp allowedcpus.out
+'
+test_expect_success 'sdexec can set unit AllowedCPUs to an empty bitmask' '
+	cat >allowedcpus2.exp <<-EOT &&
+	AllowedCPUs=
+	EOT
+	$sdexec -r 0 \
+	    --setopt=SDEXEC_NAME="t2409-allowedcpus2.service" \
+	    --setopt=SDEXEC_PROP_AllowedCPUs="" \
+	    systemctl --user show --property AllowedCPUs \
+	        t2409-allowedcpus2.service >allowedcpus2.out &&
+	test_cmp allowedcpus2.exp allowedcpus2.out
+'
+
+memtotal() {
+	local kb=$(awk '/MemTotal/ {print $2}' /proc/meminfo)
+	expr $kb \* 1024
+}
+
+# There's a rounding error in here somewhere because 50% is off by 2048 bytes,
+# but 100% is accurate with respect to /proc/meminfo.  Just use 100% for now.
+test_expect_success 'sdexec can set unit MemoryLow property' '
+	echo MemoryLow=$(memtotal) >memlow.exp &&
+	$sdexec -r 0 \
+	    --setopt=SDEXEC_NAME="t2409-memlow.service" \
+	    --setopt=SDEXEC_PROP_MemoryLow=100% \
+	    systemctl --user show --property MemoryLow \
+	        t2409-memlow.service >memlow.out &&
+	test_cmp memlow.exp memlow.out
+'
```

Comment:
two comments.

- dunno if you saw how i did tests in https://github.com/flux-framework/flux-core/pull/4255, I checked that the memory was within 1% of the expected value, so can test more options.

- Also, dunno if intentionally didn't want to test more like if cpusets set correctly?  Most of the extra testing needs cgroups2 support, so dunno if just want to deal with later?  If want to come back to later, perhaps could add comment or create issue to be done later?

Reply:
Yes I did review those tests, and yeah my idea was that we'd expand testing when we get down to using cgroups in anger.  Here I mainly just wanted to show that we can set the control properties on our transient units, not check whether that has a particular effect.  I can add a note to that effect.